### DISCUSSION TITLE
Replace global DRAGSTART and DRAGEND events with proper gesture processing

### DISCUSSION THREAD

**Discussion by parasyte** [melonjs/melonJS#1069] [Ideas]
The draggable entity is the only code that publishes the global drag events (weird! this should be handled by the lower level pointer code) The draggable entity receives its own DRAG event after handling the normal `up`/`down` event. AFAICT, It does this specifically to trigger events in the tests. It seems that doing tests would be easier with a proper API to trigger or invoke a synthetic event. On par with the [DOM `dispatchEvent` method](https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events)

Second, it might be nice if melonJS supported "real" gestures; `drag`, `pinch`, `swipe`, `double tap`, etc. Not just pretending that `DRAGSTART` and `DRAGEND` are a thing. There's a nice iOS-specific series of touch-gesture events to piggyback off of: https://developer.apple.com/library/mac/documentation/AppleApplications/Reference/SafariWebContent/HandlingEvents/HandlingEvents.html

**Comment by obiot** [comment]
About the second point :
Although I know there is not yet a standardized API for gesture, i'm not sure i'm a big fan of adding iOS specific event in the framework, when we try to propose a API that is working cross-platforms. I would rather maybe go for an emulated version of the MS one, since we already provide an cross-browser pointer-like event API (that is standardized, and on which MSGesture relies) : https://msdn.microsoft.com/en-us/library/hh968249(v=vs.85).aspx
